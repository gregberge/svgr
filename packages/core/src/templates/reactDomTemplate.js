import { getProps } from './util'

const reactDomTemplate = (code, config, state) => {
  const props = getProps(config)

  let result = `import React from 'react'\n\n`
  result += `const ${state.componentName} = ${props} => ${code}\n\n`

  if (state.webpack && state.webpack.previousExport) {
    result += `export default ${state.webpack.previousExport}\n`
    result += `export { ${state.componentName} as ReactComponent }`
  } else if (state.rollup && state.rollup.previousExport) {
    result += `${state.rollup.previousExport}\n`
    result += `export { ${state.componentName} as ReactComponent }`
  } else {
    result += `export default ${state.componentName}`
  }

  return result
}

export default reactDomTemplate
