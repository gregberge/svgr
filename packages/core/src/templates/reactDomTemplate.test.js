import reactDomTemplate from './reactDomTemplate'

describe('reactDomTemplate', () => {
  it('should wrap code into a component', () => {
    expect(
      reactDomTemplate('<Svg />', {}, { componentName: 'Test' }),
    ).toMatchSnapshot()
    expect(
      reactDomTemplate(
        '<Svg />',
        { titleProp: true },
        { componentName: 'Test' },
      ),
    ).toMatchSnapshot()
    expect(
      reactDomTemplate('<Svg />', { ref: true }, { componentName: 'Test' }),
    ).toMatchSnapshot()
    expect(
      reactDomTemplate(
        '<Svg />',
        { expandProps: true },
        { componentName: 'Test' },
      ),
    ).toMatchSnapshot()
    expect(
      reactDomTemplate(
        '<Svg />',
        { titleProp: true, ref: true },
        { componentName: 'Test' },
      ),
    ).toMatchSnapshot()
    expect(
      reactDomTemplate(
        '<Svg />',
        { titleProp: true, expandProps: true },
        { componentName: 'Test' },
      ),
    ).toMatchSnapshot()
    expect(
      reactDomTemplate(
        '<Svg />',
        { ref: true, expandProps: true },
        { componentName: 'Test' },
      ),
    ).toMatchSnapshot()
  })
})
