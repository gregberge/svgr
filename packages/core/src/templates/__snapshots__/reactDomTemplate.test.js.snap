// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reactDomTemplate should wrap code into a component 1`] = `
"import React from 'react'

const Test = () => <Svg />

export default Test"
`;

exports[`reactDomTemplate should wrap code into a component 2`] = `
"import React from 'react'

const Test = ({ title }) => <Svg />

export default Test"
`;

exports[`reactDomTemplate should wrap code into a component 3`] = `
"import React from 'react'

const Test = ({ svgRef }) => <Svg />

export default Test"
`;

exports[`reactDomTemplate should wrap code into a component 4`] = `
"import React from 'react'

const Test = props => <Svg />

export default Test"
`;

exports[`reactDomTemplate should wrap code into a component 5`] = `
"import React from 'react'

const Test = ({ svgRef, title }) => <Svg />

export default Test"
`;

exports[`reactDomTemplate should wrap code into a component 6`] = `
"import React from 'react'

const Test = ({ title, ...props }) => <Svg />

export default Test"
`;

exports[`reactDomTemplate should wrap code into a component 7`] = `
"import React from 'react'

const Test = ({ svgRef, ...props }) => <Svg />

export default Test"
`;
